hamiltonian_flow/models/harmonic_oscillator.py

```diff
@@ -0,0 +1,249 @@
+from functools import cached_property
+from typing import Dict, Literal, Optional, Union
```

Comment:
One can write `str | int` to replace `Union[str, int]`, and `int | None` to replace `Optional[int]`. I think `Optional` and `Union` can be deprecated.

Reply:
Also one can now use `dict[str, int]` directly

Reply:
> One can write `str | int` to replace `Union[str, int]`, and `int | None` to replace `Optional[int]`. I think `Optional` and `Union` can be deprecated.

This requires 3.10. We would like to include 3.9

https://github.com/kausalflow/hamiltonian_flow/blob/1c2b2021e3da58fd7a273b089593f545f4f0c750/pyproject.toml#L11

Reply:
> Also one can now use `dict[str, int]` directly

TIL.

Less imports.